docs/COREDUMP.md

```diff
@@ -0,0 +1,148 @@
+   all invoked processes, from earliest boot on. (The Linux kernel sets the
+   limit to 0 by default.)
+
+2. At the same time it will turn off coredump handling in the kernel by writing
+   `|/bin/false` into `/proc/sys/kernel/core_pattern` (see
+   [core(5)](https://man7.org/linux/man-pages/man5/core.5.html) for
+   details). This means that coredumps are not actually accepted by
+   userspace. (The linux kernel sets the pattern to `core` by default, so that
+   coredumps are written to the current working directory of the crashing
+   process.)
+
+Net effect: by default coredumps are disabled, but by means of the
+`core_pattern` rather than by size limit. This is generally preferable, since
+the pattern can be updated trivially at the right time to enable coredumping
+once the system is ready, taking comprehensive effect on all userspace. (Or to
+say this differently: disabling coredumps via the size limit is problematic,
+since it cannot easily be undone without iterating through all already running
+processes once the system is ready for coredump handling.)
+
+By updating the `core_pattern` at an appropriate point during the boot,
+coredump processing may be enabled. This moment can precisely be scheduled: any
+coredumps happening before that point are suppressed, coredumps happening after
+are processed.
+
+During the final shutdown phase the `core_pattern` is updated again, and again
+the string `|/bin/false` is written into it, disabling coredump support again,
+should it have been enabled in the meantime.
+
+This means coredump handling is generally not available during earliest boot
+and latest shutdown, reflecting the fact that storage is typically not
+available in these environments, and many other facilities are missing too that
+are required to collect and process a coredump successfully.
+
+## `systemd-coredump` Handler
+
+The systemd suite provides a coredump handler
+[`systemd-coredump`](https://www.freedesktop.org/software/systemd/man/systemd-coredump.html)
+which can be enabled at build-time. It is activated during boot via the
+`/usr/lib/sysctl.d/50-coredump.conf` drop-in file for
+`systemd-sysctl.service`. It registers the `systemd-coredump` tool as
+`core_pattern`.
+
+`systemd-coredump` is implemented as socket activated service: when the kernel
+invokes the userspace coredump handler the received serialization file
+descriptor is handed off to the socket activated service
+`systemd-coredump@.service` via the `systemd-coredump.socket` socket unit. This
+means the coredump handler runs for a very short time only, and the potentially
+*heavy* and security sensitive coredump processing work is done as part of the
+specified service unit, and thus can take benefit of regular service resource
+management and sandboxing.
+
+The `systemd-coredump` handler will extract a backtrace from any coredumps it
+receieves, which it will then log to the journal. Coredumps thus automatically
```

Comment:
"which it will then log" → "and log it"

Strike "Coredumps thus automatically become a loggable event.". That's some strange jargon.

"Futher analysis of coredumps" → "The information about coredumps stored in the journal can be enumerated and queried" or something like that? `coredumpctl` doesn't really do any "analysis".